tests/test_validators.py

```diff
@@ -0,0 +1,79 @@
+"""This is an example of json schema for the GDC using schemas defined
+in local yaml files (see gdcdictionary/examples).
+
+This test class is making sure the example jsons comply with the schema.
+
+Note this is NOT testing that the schema is sane. Just that we adhere
+to it
+
+"""
+try:
+    from importlib.resources import files
+except ImportError:
+    from importlib_resources import files
+
+import json
+import pytest
+
+import gdcdictionary
+
+
+def get_all_paths(subdir: str) -> str:
+    with files("tests") as path:
+        examples_path = path.parent / f"examples/{subdir}"
+        yield from sorted(examples_path.glob("*.json"))
+
+
+@pytest.mark.parametrize("path", get_all_paths("valid"))
+def test_valid_examples(path, schema):
+    with open(path) as f:
+        doc = json.load(f)
+        assert len(gdcdictionary.validate(doc)) == 0
+
+
+@pytest.mark.parametrize("path", get_all_paths("invalid"))
+def test_invalid_examples(path, schema):
+    with open(path) as f:
+        doc = json.load(f)
+        violations = gdcdictionary.validate(doc)
+        assert len(violations) > 0
+        print(violations)
+
+
+def test_validate_instances():
+    paths = get_all_paths("valid")
+    instances = []
+    for path in paths:
+        with open(path) as f:
+            doc = json.load(f)
+            instances.append(doc)
+    violations = gdcdictionary.validate_instances(instances)
+    assert len(violations) == 0
+
+
+def test_validate_instances__invalid_types():
```

Comment:
To keep the focus of the tests narrow, an optional suggestion would be to split this into multiple tests (one for unknown types, one for missing required and one for extra fields).

Reply:
I ended up leaving the test as is and added log for the violations. The reason is that, this test is essentially the same as `test_invalid_examples` but its testing using a bulk json. The violations I asserted are a subset of the resulting errors but they form the three scenarios in the input. These are not all possible combination of violations, they are just the three in the sample input. Splitting this into three makes it look like these 3 are more important than others, which they are not and I cannot write tests for all the possible combinations.
``` python
 [
        SchemaValidationError(
            schema="",
            message="specified type: species is not in the current data model",
            keys=["type"],
        ),
        SchemaValidationError(
            schema="case",
            message="Key(s) ['python_version'] not a valid property for type 'case'",
            keys=["python_version"],
        ),
        SchemaValidationError(
            schema="case",
            message="'submitter_id' is a required property",
            keys=["submitter_id"],
        ),
        SchemaValidationError(
            schema="case",
            message="'disease_type' is a required property",
            keys=["disease_type"],
        ),
        SchemaValidationError(
            schema="case",
            message="'primary_site' is a required property",
            keys=["primary_site"],
        ),
        SchemaValidationError(
            schema="", message="'type' is a required property", keys=["type"]
        ),
    ]
```

Reply:
Ah understood, that makes a lot of sense, that we want to test a few combinations but not an exhaustive test. Thank you, do you mind adding a docstring to make the intention clear to future maintainers?